CarEshipApp/app/src/main/java/com/example/careshipapp/gui/activities/CustomerPaymentActivity.java

```diff
@@ -1,134 +1,228 @@
+                                                    }
+
+                                                }) .addOnFailureListener(new OnFailureListener() {
+
+                                                    @Override
+                                                    public void onFailure(@NonNull Exception e) {
+
+                                                        Toast.makeText(CustomerPaymentActivity.this,"Something went wrong.",Toast.LENGTH_SHORT).show();
+
+                                                    }
+                                                });
+
+                                    }
+                                }
+
+                            });
+
+                }
+
+            }
+            private String currentDate()
+            {
+                java.util.Calendar cal = java.util.Calendar.getInstance();
+                int year = cal.get(java.util.Calendar.YEAR);
+                int month = cal.get(java.util.Calendar.MONTH);
+                month = month + 1;
+                int day = cal.get(java.util.Calendar.DAY_OF_MONTH);
+                return dateString(day, month, year);
+            }
+
+            private void dateShow(){
+                DatePickerDialog.OnDateSetListener dateSetListener = new DatePickerDialog.OnDateSetListener() {
                     @Override
-                    public void onClick(View view) {
-                        if(!user.isEmailVerified()) {
-                            verifyMsg.setVisibility(View.VISIBLE);
-                            Toast.makeText(CustomerPaymentActivity.this,"Email is not verified yet",Toast.LENGTH_SHORT).show();
-                        } else {
-                           Intent intent = new Intent(CustomerPaymentActivity.this, PaymentConfirmationActivity.class);
-                        }
+                    public void onDateSet(DatePicker datePicker, int year, int month, int day) {
+                        String date = dateString(day, month, year);
+                        month = month + 1;
+                        expiryDatePicker.setText(date);
+
                     }
-                });
-            }
-        });
+                };
 
+                java.util.Calendar calender = java.util.Calendar.getInstance();
+                int year = calender.get(java.util.Calendar.YEAR);
+                int month = calender.get(java.util.Calendar.MONTH);
+                int day = calender.get(Calendar.DAY_OF_MONTH);
 
+                dialog = new DatePickerDialog(this, dateSetListener, year, month, day);
 
-        //Intent intent = new Intent(CustomerPayment.this,EmailVerification.class);
 
 
+            }
 
-        }
+            private String dateString(int day, int month, int year) {
+                return   day + "-" + monthFormat(month) + "-" + year;
+            }
 
-        }
 
 
+            private String monthFormat(int month){
```

Comment:
Can be substituted with a switch statements, I believe they are faster in execution than many ifs.

Reply:
Okay, nevermind:)